Foundation/NSStream.swift

```diff
@@ -154,17 +205,97 @@ public class InputStream: Stream {
         CFReadStreamClose(_stream)
     }
     
+    public override var streamError: NSError?{
+        let error = CFReadStreamCopyError(_stream)
+        return error?._nsObject
+    }
+    
     public override var streamStatus: Status {
-        return Stream.Status(rawValue: UInt(CFReadStreamGetStatus(_stream)))!
+        return NSStream.Status(rawValue: UInt(CFReadStreamGetStatus(_stream)))!
+    }
+    
+    public override func setProperty(_ property: AnyObject?, forKey key: String) -> Bool {
+        return CFReadStreamSetProperty(_stream, key._cfObject, property)
+    }
+    
+    public override func propertyForKey(_ key: String) -> AnyObject? {
+        return CFReadStreamCopyProperty(_stream, key._cfObject)
     }
+    
+    public func copy(with zone: NSZone?) -> AnyObject{
```

Comment:
Not sure caching the url and data to get away with copy and mutableCopy is a good idea.
If any one has options let me know